gensim/test/test_fasttext.py

```diff
@@ -1007,52 +958,64 @@ def test_out_of_vocab(self):
             actual_vector = native.wv.word_vec(word)
             self.assertTrue(np.allclose(expected_vector, actual_vector, atol=1e-5))
 
+        self.model_structural_sanity(native)
+
     @unittest.skip('this test does not pass currently, I suspect a bug in our FT implementation')
     def test_out_of_vocab_gensim(self):
         """Test whether gensim gives similar results to FB for OOV words.
 
         Seems to be broken for our toy model.
+
+        # GM: probably unreasonable to expect identical results given alg randomization & thread jitter
```

Comment:
So test without threads? What other sources of randomization are there, we are using the same hashing right?

Reply:
The skip note above may have been an early warning of he wrongness of our early FT implementation, and the choice to skip the test rather than investigate caused many hours/days of pain! But on the other hand, matching every calc across (languages/libraries/threads/PRNGs) could be a lot of time-consuming work that fails, or an arduous temporary success whose stability is discarded as soon as we take other necessary steps (like using locally-efficient randomization in initialization, rather than identical-to-FB-FT-C++-PRNGs). So I wouldn't recommend ensuring that level of identicality after training. Just (1) similar quality of results, on generally equivalent runs; (2) essentially-identical word/OOV-word calculations, when loading an already-trained/frozen model.